lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -233,6 +258,21 @@ public T addNodeName(final String nodeName) {
     }
 
     /**
+     * Adds node names.They are the preferred locations where the evaluator should
+     * run on. If any of the node is available, the RM will try to allocate the
+     * evaluator there
+     *
+     * @param nodeNamesList preferred node names
+     * @return this Builder.
+     */
+    public T addNodeNames(final List<String> nodeNamesList) {
+      if(nodeNamesList != null) {
```

Comment:
When we create ContainerRequest, if nodeNameList is empty, we pass null for nodes. Test shows without passing any node names, the code still works.